### ISSUE TITLE
Wrong path name when installing packages on Git Bash for Windows(zsh shell)

### ISSUE THREAD

**Issue by benwilburn** [mattmc3/antidote#121]
When attempting to install packages via antidote in Git Bash for Windows, it fails to install packages due to path errors (seems to work fine in WSL). It would be nice to be able to use Antidote in Git Bash

**Comment by mattmc3** [comment]
Git Bash is, by default, Bash. I hate to ask, but can you confirm you are you running Zsh from bash before running antidote, or are you just hoping antidote will work in straight git bash? Antidote is, and will always be, a Zsh-only plugin manager, so I just need to confirm.

**Comment by benwilburn** [comment]
Yes, I even mentioned it in the title of the issue. I have used oh-my-zsh in git bash (ash shell) but it is brutally slow.

When I echo the $SHELL variable, it says bash but I have it set to open zsh upon load.

**Comment by benwilburn** [comment]
![image](https://user-images.githubusercontent.com/18269925/232171694-2cf08eff-a4b0-4af8-ac7b-5f9072b24792.png)

**Comment by RotaN8** [comment]
> Yes, I even mentioned it in the title of the issue. I have used oh-my-zsh in git bash (zsh shell) but it is brutally slow.
> 
> When I echo the $SHELL variable, it says bash but I have it set to open zsh upon load.
> 
> ![image](https://user-images.githubusercontent.com/18269925/232171704-1bc493cc-c149-4715-9e59-8db2c7d5b7a9.png)

You can ask for solutions on this GitHub Gist (maybe):
https://gist.github.com/fworks/af4c896c9de47d827d4caa6fd7154b6b

But if you are unable to be running Zsh before `antidote load` is executed obviously it will fail. And it maybe fail anyway because it's a weird setup using Git BASH (his name says everything...) but trying to run  Zsh in it. Also Windows may be interfering with it because git bash is not a perfect recreation of a Linux shell and FS. Stuff done for Zsh won't work as expected, I think this is obvious, isn't it?

In my opinion this is not a problem that the antidote developer (@mattmc3) should help troubleshooting because the used setup is not the pretended to be.

And as you said, you could always use WSL that is (by far) the best solution (and a native) to run Linux stuff on Windows. And you can do everything you do with Git Bash with WSL but in a more efficient way and without almost any problem because in the back it is a Linux system.

That being said I hope you could get this to work as you wish to.

I'll recommend you to use WSL or (like me) a dual-boot set up with an entire Linux distro paired with your Windows. And you will be able to do whatever you want with the PC 😉 I think is the best setup today taking into count that the price of the SSD are so low and they are big enough to carry two SO.

PD: I'm not part of the development team I'm just a random user of this awesome plugin manager. And my answer is not pretended to be mean, I'm only saying the things as-is (or almost how I think they are).

**Comment by mattmc3** [comment]
I'm open to see if we can get things working in git bash. I just don't have a Windows machine handy at the moment, so it'll take me a bit to get a test scenario set up to see if I can reproduce this. If so, and it's something that can be addressed within antidote, I'll consider coding a fix.

**Comment by DiegoMagdaleno** [comment]
Without any promises I am able to use windows and have some experience with shell programming.

I am on finals but I am pretty sure this is an easy fix.

I'll see if I can fix the issue and submit a pr :)

**Comment by DiegoMagdaleno** [comment]
One question to speed things up. Per my understanding it seems like the path used when antidote generates `zsh_plugins.zsh` is `$ANTIDOTE_HOME`?

It would be helpful to know this information, as I can just make it point to the right location when on Mingw64

**Comment by mattmc3** [comment]
This is the function that determines the home: https://github.com/mattmc3/antidote/blob/main/functions/antidote-home

**Comment by DiegoMagdaleno** [comment]
Submited a PR!

**Comment by benwilburn** [comment]
@mattmc3 I just tried to set it up again and am getting the same issue:
![image](https://github.com/mattmc3/antidote/assets/18269925/6cd1072c-ee07-4ef8-9961-05f3c5891e83)

**Comment by DiegoMagdaleno** [comment]
@benwilburn can you try using my fork and see if that works?

**Comment by marovira** [comment]
I'm also having the same issue as @benwilburn when trying to use antidote in zsh invoked from git bash. If I add the `__fix_windows_path` function to `functions/__antidote_get_cachedir` and use it like this:
```
  local result
  if [[ "${OSTYPE}" == darwin* ]]; then
    result=$HOME/Library/Caches
  elif [[ "${OSTYPE}" == (cygwin|msys)* ]]; then
    result=$(__fix_windows_path ${LOCALAPPDATA:-$LocalAppData}) # <--- Note the change for the path
  elif [[ -n "$XDG_CACHE_HOME" ]]; then
    result=$XDG_CACHE_HOME
  else
    result=$HOME/.cache
  fi
```

Then the issue is fixed and antidote works correctly. Could the fix by @DiegoMagdaleno be merged in?

**Comment by mattmc3** [comment]
I can't merge that in without a good way to test it into the future, so what I'm really held up by is making sure there's a CI workflow to verify this works. I don't run Windows.

**Comment by marovira** [comment]
There may be a way to have the GitHub runners handle this, since they do come with git bash installed by default. @mattmc3, if I figure out a way to hook it into a Windows runner, would you be open to a PR?

**Comment by mattmc3** [comment]
Yes, definitely.

**Comment by DiegoMagdaleno** [comment]
I can take full responsability for maintaining all of the Windows related bugs.

**Comment by mattmc3** [comment]
If v1.9.2 doesn't fix it, please re-open. I was able to properly test on an old Windows clunker, so I'm reasonably confident I can support this without CI working for Windows yet. No reason to hold up the feature. Thanks for the `cygpath` implementation @marovira.
